worlds_nif/nif-portugal.py

```diff
@@ -0,0 +1,143 @@
+from random import randint
+
+# FORMATTING
+#############
+
+
+def remove_symbols(dirty_nif):  # type: (str) -> str
+    """
+    Removes spaces, dots, and other symbols from a NIF.
+
+    Args:
+        dirty_nif (str): A string containing a NIF with possible non-numeric characters.
+
+    Returns:
+        str: A cleaned NIF containing only numeric digits.
+
+    Example:
+        remove_symbols("123.456.789") -> "123456789"
+    """
+    return "".join(filter(str.isdigit, dirty_nif))
+
+
+def format_nif(nif):  # type: (str) -> str
+    """
+    Formats a Portuguese NIF for display with visual aid symbols.
+
+    This function formats a numbers-only NIF string by adding periods and
+    a dash for better readability, commonly used in Portuguese documents.
+
+    Args:
+        nif (str): A numbers-only NIF string of exactly 9 digits.
+
+    Returns:
+        str: A formatted NIF string with periods and a dash, or None
+        if the input is invalid.
+
+    Example:
+        format_nif("123456789") -> "123.456.789-0"
+    """
+    # Clean the NIF first, in case it's with any symbols
+    nif = remove_symbols(nif)
+
+    # Ensure the NIF is valid for formatting
+    if len(nif) != 9:
+        return None
+
+    # Add periods after the third and sixth digits, and a dash before the last digit
+    return "{}.{}.{}-{}".format(nif[:3], nif[3:6], nif[6:8], nif[8])
+
+
+# OPERATIONS
+#############
+
+
+def validate(nif):  # type: (str) -> bool
```

Comment:
Mesmo esquema aqui, mudar o nome do método para is_valid:

```suggestion
def is_valid(nif):  # type: (str) -> bool
```